extensions/Worker.Extensions.Tables/src/Config/TablesBindingOptions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Azure.Core;
+using Azure.Data.Tables;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Tables.Config
+{
+    internal class TablesBindingOptions
+    {
+        public string? ConnectionString { get; set; }
+
+        public Uri? ServiceUri { get; set; }
+
+        public TokenCredential? Credential { get; set; }
+
+        public TableClientOptions? TableClientOptions { get; set; }
+
+        private TableServiceClient? TableServiceClient { get; set; }
```

Comment:
For `private`, I typically use fields (as opposed to properties). But no harm in keeping it a property.